components/eamxx/src/share/property_checks/mass_and_energy_column_conservation_check.cpp

```diff
@@ -278,6 +283,152 @@ PropertyCheck::ResultAndMsg MassAndEnergyColumnConservationCheck::check() const
+  const auto qr             = m_fields.at("qr"            ).get_view<const Real**> ();
+  const auto ps             = m_fields.at("ps"            ).get_view<const Real*>  ();
+  const auto phis           = m_fields.at("phis"          ).get_view<const Real*>  ();
+
+  const auto vapor_flux = m_fields.at("vapor_flux").get_view<const Real*>();
+  const auto water_flux = m_fields.at("water_flux").get_view<const Real*>();
+  const auto ice_flux   = m_fields.at("ice_flux"  ).get_view<const Real*>();
+  const auto heat_flux  = m_fields.at("heat_flux" ).get_view<const Real*>();
+
+  const auto policy = ExeSpaceUtils::get_default_team_policy(ncols, nlevs);
+
+  using namespace ekat::units;
+  using namespace ShortFieldTagsNames;
+
+  //better to get layouts from m_grid, even scalar0d
+  FieldLayout scalar0d({},{});
+  FieldIdentifier s0_fid("s0", scalar0d, kg / kg, m_grid->name());
+  Field res(s0_fid);
+  res.allocate_view();
+  //this does not exist, why? cause it can be a LEV or a COL array?
+  //FieldLayout scalar1d = m_grid->get_1d_scalar_layout();
+  FieldLayout scalar1d{{COL}, {m_num_cols}};
+  FieldIdentifier s1_fid("s1", scalar1d, kg / kg, m_grid->name());
+
+  Field field_version_s1(s1_fid);
+  field_version_s1.allocate_view();
+  auto field_view_s1 = field_version_s1.get_view<Real*>();
+
+  auto area = m_grid->get_geometry_data("area").clone();
+  auto area_view = area.get_view<const Real*>();
+
+  Kokkos::parallel_for(policy, KOKKOS_LAMBDA (const KT::MemberType& team) {
+    const int i = team.league_rank();
+    const auto pseudo_density_i = ekat::subview(pseudo_density, i);
+    // Calculate total gas mass (sum dp, no water loading)
+    field_view_s1(i) = compute_gas_mass_on_column(team, nlevs, pseudo_density_i) * area_view(i);
+  });
+
+  //reprosum vars
+  const Real* send;
+  Real* recv;
+  Int nlocal = ncols;
+  Int ncount = 1;
+  auto s1_host = field_version_s1.get_view<const Real*,Host>();
+  auto s0_host = res.get_view<Real,Host>();
+  send = s1_host.data();
+  recv = s0_host.data();
+
+  eamxx_repro_sum(send, recv, nlocal, ncount, MPI_Comm_c2f(m_comm.mpi_comm()));
+  total_gas_mass_after = (*recv);
+
+//this ||4 needs to be 2 for-loops, with one summing each 4 cols first, serially
+//for pg2 grids (if on np4 grids, it would require much more work?)  
+  Kokkos::parallel_for(policy, KOKKOS_LAMBDA (const KT::MemberType& team) {
+
+    const int i = team.league_rank();
+    const auto pseudo_density_i = ekat::subview(pseudo_density, i);
+    const auto T_mid_i          = ekat::subview(T_mid, i);
+    const auto horiz_winds_i    = ekat::subview(horiz_winds, i);
+    const auto qv_i             = ekat::subview(qv, i);
+    const auto qc_i             = ekat::subview(qc, i);
+    const auto qr_i             = ekat::subview(qr, i);
+    const auto qi_i             = ekat::subview(qi, i);
+
+    // Calculate total energy
+    m_new_energy_for_fixer(i) = compute_total_energy_on_column(team, nlevs, pseudo_density_i, T_mid_i, horiz_winds_i,
```

Comment:
This value of m_new_energy_for_fixer doesn't seem to be used anywhere else. It can just be a local scalar within this kernel, it appears, and the same for the other use.

Reply:
thanks, done